service/src/main/java/bio/terra/externalcreds/services/SshKeyPairService.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.externalcreds.services;
+
+import bio.terra.common.db.ReadTransaction;
+import bio.terra.common.db.WriteTransaction;
+import bio.terra.externalcreds.dataAccess.SshKeyPairDAO;
+import bio.terra.externalcreds.generated.model.SshKeyPairType;
+import bio.terra.externalcreds.models.SshKeyPair;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SshKeyPairService {
+
+  private final SshKeyPairDAO sshKeyPairDAO;
+
+  public SshKeyPairService(SshKeyPairDAO sshKeyPairDAO) {
+    this.sshKeyPairDAO = sshKeyPairDAO;
+  }
+
+  @ReadTransaction
```

Comment:
What is the benefit of having the transaction annotations here instead of in the DAO?

Reply:
i copied this pattern from other services class.

Reply:
The pattern this service follows is transaction demarcation at the service level to keep DAOs free from business logic.